src/drivers/intel/cavs/dmic.c

```diff
@@ -832,6 +805,47 @@ static inline void ipm_helper2(int source[], int *ipm, int stereo[],
 }
 #endif
 
+/* Loop number of PDM controllers in the configuration. The function
+ * checks if the controller should operate as stereo or mono left (A)
+ * or mono right (B) mode. Mono right mode is setup as channel
+ * swapped mono left.
+ */
+static int stereo_helper(int stereo[], int swap[], int di)
+{
+	int cnt;
+	int i;
+	int swap_check;
+	int ret = 0;
+
+	for (i = 0; i < DMIC_HW_CONTROLLERS; i++) {
+		cnt = 0;
+		if (dmic_prm[0]->pdm[i].enable_mic_a ||
+		    dmic_prm[1]->pdm[i].enable_mic_a)
+			cnt++;
+
+		if (dmic_prm[0]->pdm[i].enable_mic_b ||
+		    dmic_prm[1]->pdm[i].enable_mic_b)
+			cnt++;
+
+		/* Set stereo mode if both mic A anc B are enabled. */
+		cnt >>= 1;
+		stereo[i] = cnt;
+
+		/* Swap channels if only mic B is used for mono processing. */
+		swap[i] = dmic_prm[di]->pdm[i].enable_mic_b & !cnt;
+
+		/* Check that swap does not conflict with other DAI request */
+		if (!di)
+			swap_check = dmic_prm[1]->pdm[i].enable_mic_a;
+		else
+			swap_check = dmic_prm[0]->pdm[i].enable_mic_a;
+
+		if (swap_check && swap[i])
```

Comment:
Break the loop?

Reply:
It would leave the stereo and swap arrays processing incomplete and add complexity & code size. I'd keep it as is since it's in most platforms just two iterations.